diff --git a/node_modules/reveal-md/lib/listing.js b/node_modules/reveal-md/lib/listing.js
index 533c38d..f4b65e6 100644
--- a/node_modules/reveal-md/lib/listing.js
+++ b/node_modules/reveal-md/lib/listing.js
@@ -28,7 +28,7 @@ const renderListFile = async filePaths => {
   const { title, listingTemplate, theme, assetsDir } = getOptions();
   const template = await getListingTemplate(listingTemplate);
   const themeUrl = getThemeUrl(theme, assetsDir, '.');
-  const files = await Promise.all(filePaths.map(getFileMeta));
+  const files = await Promise.all(filePaths.reverse().map(getFileMeta));
   return Mustache.render(template, {
     base: '',
     themeUrl,
